### ISSUE TITLE
(Windows) Creating both a rodio device and a winit window causes a panic

### ISSUE THREAD

**Issue by Raz-Hemo** [RustAudio/rodio#270]
The panic happens either if you create the device first and window second, or the other way around, effectively meaning you cannot use rodio and winit together.
How to reproduce:

```
...
eventloop.run(move |event, _, control_flow| {
        *control_flow = ControlFlow::Poll;

        match event {
            Event::NewEvents(winit::event::StartCause::Init) => {
                let device = rodio::default_output_device().unwrap();
            }
...
```

(which panics with "src\libcore\result.rs:1165: called `Result::unwrap()` on an `Err` value: Os { code: -2147417850, kind: Other, message: "Cannot change thread mode after it is set." }")

OR
```
...
let device = rodio::default_output_device().unwrap();
eventloop.run(move |event, _, control_flow| {
...
```
(which panics with "github.com-1ecc6299db9ec823\winit-0.21.0\src\platform_impl\windows\window.rs:82: OleInitialize failed! Result was: `RPC_E_CHANGED_MODE`")

Am I missing something? Do I need to create a separate thread exclusively for rodio?

**Comment by Imberflur** [comment]
This is caused by the issue mentioned here https://github.com/rust-windowing/winit/issues/1255 and here https://github.com/RustAudio/cpal/pull/348

Currently, the only workaround that I have found is using rodio from a separate thread. If you are making infrequent calls to rodio and don't want to create a dedicated separate thread one thing I have done is create a short lived thread:
```rust
let device = std::thread::spawn(|| rodio::default_output_device().unrwap())
    .join()
    .unwrap();
```

**Comment by Rua** [comment]
That workaround doesn't seem to work for me. It avoids an error on initialisation, but then I get the error later as soon as I try to play a sound.

**Comment by paulirotta** [comment]
@Rua the long-lived-own-thread solution may work better than the short lived thread example.

The issue may be Winit and Cpal not agreeing on how they initialize the thread state. See https://docs.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-coinitializeex

**Comment by lukors** [comment]
Is this merged PR a fix for this? https://github.com/RustAudio/cpal/pull/597

**Comment by est31** [comment]
Closing because it looks a lot like the bug that the PR that @lukors linked above fixed. It's included in cpal 0.13.4. If anyone still sees the bug please comment and I'll reopen.
